config/routes.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users', to: 'users#index'
+  get 'users/:id', to: "users#show", as: 'user'
+  get 'users/:author_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:author_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- [ ] We are taking the approach of nested routing, so that means on lines 6 and 7 we should have `user_id`, not `author_id`. Kindly fix that.

  SUGGESTION: Instead of this lengthy route definition we can take a more concise approach, like so:
  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

Reply:
Noted, i have changed updated the name of the id